app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -146,6 +149,13 @@ export default class Heatmap {
     this.processData("parse");
   }
 
+  updateCaption(caption) {
+    this.options.printCaption = caption;
+    // We need to call placeContainers() to resize the height of the heatmap based on the number of captions.
+    // And update() to actually change the captions.
+    this.processData("placeContainers");
```

Comment:
We should not force the whole update function just for adding captions that might not even be necessary (if user does not trigger download).